library/Vanilla/DateFilterSchema.php

```diff
@@ -160,16 +189,27 @@ public function parseSimple($date, $operator, ValidationField $field) {
      * {@inheritdoc}
      */
     public function validate($data, $sparse = false) {
-        $field = new ValidationField($this->createValidation(), $this->getSchemaArray(), '', $sparse);
+        $validation = $this->createValidation();
+        $field = new ValidationField($validation, $this->getSchemaArray(), '', $sparse);
+
+        if (is_string($data)) {
+            $data = $this->stringToSchema($data, $field);
+        }
 
-        $clean = $this->validateDateFilter($data, $field);
+        if (Invalid::isValid($data)) {
+            if (is_array($data)) {
+                $clean = $this->validateField($data, $field, $sparse);
```

Comment:
Nesting the assignment of `$clean` beneath two conditions means there's the possibility it won't be set, even though it's the function's return value.

Reply:
In that case an exception will be thrown so it doesn't matter.